packages/teleterm/src/ui/DocumentCluster/ClusterResources/Servers/Servers.tsx

```diff
@@ -114,3 +153,10 @@ const renderAddressCell = ({ addr, tunnel }: types.Server) => (
     {!tunnel && addr}
   </Cell>
 );
+
+const Wrapper = styled(Box)`
+  &.disabled {
+    pointer-events: none;
+    opacity: 0.5;
+  }
```

Comment:
I finally got around to testing actual pagination, the following patch gives me "infinite" pagination to play around with.

<details>
<summary>patch</summary>

```diff
diff --git a/lib/teleterm/apiserver/handler/handler_servers.go b/lib/teleterm/apiserver/handler/handler_servers.go
index 685f6ee6d..c37fd70e1 100644
--- a/lib/teleterm/apiserver/handler/handler_servers.go
+++ b/lib/teleterm/apiserver/handler/handler_servers.go
@@ -16,7 +16,9 @@ package handler
 
 import (
 	"context"
+	"fmt"
 	"sort"
+	"strconv"
 
 	"github.com/gravitational/trace"
 
@@ -41,18 +43,43 @@ func (s *Handler) GetAllServers(ctx context.Context, req *api.GetAllServersReque
 
 // GetServers accepts parameterized input to enable searching, sorting, and pagination
 func (s *Handler) GetServers(ctx context.Context, req *api.GetServersRequest) (*api.GetServersResponse, error) {
+	originalStartKey := req.StartKey
+	var base int
+	var err error
+	if originalStartKey == "" {
+		base = 0
+	} else {
+		base, err = strconv.Atoi(originalStartKey)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		base++
+	}
+	req.StartKey = ""
 	resp, err := s.DaemonService.GetServers(ctx, req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	var startKey int
+
 	response := &api.GetServersResponse{
-		TotalCount: int32(resp.TotalCount),
+		TotalCount: int32(base + 30),
 		StartKey:   resp.StartKey,
 	}
 	for _, srv := range resp.Servers {
-		response.Agents = append(response.Agents, newAPIServer(srv))
+		for i := 0; i < 15; i++ {
+			k := i + base
+			apiSrv := newAPIServer(srv)
+			apiSrv.Uri = fmt.Sprintf("%v%d", apiSrv.Uri, k)
+			apiSrv.Name = fmt.Sprintf("%v%d", apiSrv.Name, k)
+			apiSrv.Hostname = fmt.Sprintf("%v%d", apiSrv.Hostname, k)
+			apiSrv.Addr = fmt.Sprintf("%v%d", apiSrv.Addr, k)
+			response.Agents = append(response.Agents, apiSrv)
+			startKey = k
+		}
 	}
+	response.StartKey = strconv.Itoa(startKey)
 
 	return response, nil
 }
```
</details>

I think the transition here from no opacity to 0.5 is a bit too sudden. It causes heavy flashing when testing locally but that's also partly because I'm running the cluster locally so the requests come back almost immediately. But even after adding 250–850ms of artificial delay to the requests, it was a bit too jarring because the change in brightness is quite big and happens too fast.

What would you say about this? This needs to be repeated for `<SearchPanel>` as well.

```suggestion
  // The timing functions of transitions have been chosen so that the element loses opacity slowly
  // when entering the disabled state but gains it quickly when going out of the disabled state.
  transition: opacity 150ms ease-out;

  &.disabled {
    pointer-events: none;
    opacity: 0.7;
    transition: opacity 150ms ease-in;
  }

```

Reply:
This does look really nice